lib/arrow/adjustment.ex

```
@@ -33,6 +33,6 @@ defmodule Arrow.Adjustment do
     adjustment
     |> cast(attrs, [:source, :source_label, :route_id])
     |> validate_required([:source, :source_label, :route_id])
-    |> unique_constraint(:source_source_label, name: :adjustments_source_source_label)
+    |> unique_constraint(:source_label, name: :adjustments_source_label)
```

You don't need to specify the name here anymore, either.

---

So interestingly here if I remove the name it actually changes the behavior - it throws an exception when the constraint is violated rather than returning an `{:error, _}` tuple. I'm not sure if that's actually what we want to do.

---

Hmm... that seems very odd to me. Did you run the migration down and up on the test DB? I see [this question](https://elixirforum.com/t/unique-constraint-i-dont-know-how-to-use-name/16339) which seems to imply that the behavior differs not on whether or not `:name` is included, but if it matches a DB constraint. So I'm wondering if the test DB migrations are consistent with the schema.

---

Ah, you're right, I tried just doing `MIX_ENV=test mix ecto.drop` and then recreating the DB and running the migration and it all worked.